src/shaders/slipmodeshader.cpp

```diff
@@ -0,0 +1,60 @@
+#include "shaders/slipmodeshader.h"
+
+using namespace mixxx;
+
+void SlipModeShader::init() {
+    QString vertexShaderCode = QStringLiteral(R"--(
+#version 150
+attribute highp vec4 position; // use vec4 here (will be padded) to assign directly to gl_Position
+
+out highp vec4 vposition;
+
+void main()
+{
+    vposition = position;
+    gl_Position = position;
+}
+)--");
+
+    QString fragmentShaderCode = QStringLiteral(R"--(
+#version 120
+uniform highp vec4 color;
+uniform highp vec2 borders;
+uniform highp vec2 dimension;
+
+in highp vec4 vposition;
+
+void main()
+{
+
+    float xBorder = abs(dimension.x*vposition.x);
+    float yBorder = dimension.y*vposition.y;
```

Comment:
Not sure why clang-format is not complaining:
```suggestion
    float xBorder = abs(dimension.x * vposition.x);
    float yBorder = dimension.y * vposition.y;
```

Reply:
I was wondering that too since, I have copied this formatting in another waveform file. Perhaps it is somehow considered valid?

Reply:
I think this because it's part of the GLSL fragment shader so not parsed by clang-format